org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DefaultVariableProvider.java

```diff
@@ -76,20 +78,24 @@ private void initialize() {
 
         registerValueFormatter(new StringObjectFormatter(), 2);
         registerValueFormatter(new ArrayObjectFormatter(this::typeToString), 2);
-        registerValueFormatter(new ClassObjectFormatter(this::typeToString), 2);
+        registerValueFormatter(new ClassObjectFormatter(this::typeToString), 2);        
+        
     }
-
-    private static IFormatter getFormatter(Map<? extends IFormatter, Integer> formatterMap, Type type,
-                                           Map<String, Object> options) {
-        List<? extends IFormatter> formatterList =
-                formatterMap.keySet().stream().filter(t -> t.acceptType(type, options))
-                        .sorted((a, b) ->
-                                -Integer.compare(formatterMap.get(a), formatterMap.get(b))).collect(Collectors.toList());
-        if (formatterList.isEmpty()) {
-            throw new UnsupportedOperationException(String.format("There is no related formatter for type %s.",
-                    type == null ? "null" : type.name()));
+    
+    @Override
+    public Map<String, Object> getDefaultOptions() {
+        if (defaultOptions == null) {
```

Comment:
Suggest raising exception when defaultOptions.size() != valueFormatterMap.size() + typeFormatterMap.size();
This indicates duplication.